roles/roles_common.js

```diff
@@ -1231,6 +1231,52 @@ _.extend(Roles, {
   },
 
   /**
+   * By this method you can find out if a role is parent of another role.
+   *
+   * WARNING: If you check this on the client, please make sure all roles are published.
+   *
+   * @method isParentOf
+   * @param {String} roleName The role you have
+   * @param {String} demandedRoleName The role you want to find within roleName
+   * @private
+   * @static
+   */
+  isParentOf: function (roleName, demandedRoleName) {
+    Roles._checkRoleName(roleName);
+    Roles._checkRoleName(demandedRoleName);
+
+    if (roleName === null || demandedRoleName === null) {
+      return false;
+    }
+
+    if (roleName === demandedRoleName) {
+      return true;
+    }
+
+    var alreadyCheckedRoles = [];
```

Comment:
I do not think you have to remember this because we assume there are no cycles in the database.